### DISCUSSION TITLE
Support `less` for compatibility ?

### DISCUSSION THREAD

**Discussion by AM-young-fun** [vercel/next.js#49665] [Ideas]
### Goals

In `NextJS 13.4`, it is impossible to use `next-with-less` to support less anymore.
Can we support this by making `regexLikeCss` editable?

https://github.com/vercel/next.js/blob/f169c32d3f48ceb92abf6197226010d288cc0483/packages/next/src/build/webpack/config/blocks/css/index.ts#L18

### Background

Many projects still use Less Loader to process CSS. Wish we could have `Less` support for compatibility.
It's very painful to migrate thousands of `less` files to `sass`.

### Proposal
I want `regexLikeCSS` in Webpack build config to be configurable.

As far as I know, the following code use the `regexLikeCSS` variable:

https://github.com/vercel/next.js/blob/f169c32d3f48ceb92abf6197226010d288cc0483/packages/next/src/build/webpack/config/blocks/css/index.ts#LL218C23-L218C23

To achieve the goal, we can modify it into this:

```typescript
  // CSS cannot be imported in _document. This comes before everything because
  // global CSS nor CSS modules work in said file.
  fns.push(
    loader({
      oneOf: [
        markRemovable({
          test: nextConfig.regexLikeCSS || regexLikeCss, // >>>>>>>make it configurable here!!<<<<<<<<<
          // Use a loose regex so we don't have to crawl the file system to
          // find the real file name (if present).
          issuer: /pages[\\/]_document\./,
          use: {
            loader: 'error-loader',
            options: {
              reason: getCustomDocumentError(),
            },
          },
        }),
      ],
    })
  )

```

That reads `regexLikeCSS`'s value from `next.config` first,
if undefined then fallback to the original value as default.

After such a patch, we can support LessJS in `next.config.js` as following:

```
const withLess = require('next-with-less');

module.exports = withLess({
  regexLikeCSS: /\.(css|scss|sass|less)$/,
})
```

**Comment by nike1v** [comment]
Please, support that files into the next app.
